### PR TITLE
2756 - Account Selecotor Updates

### LINKED ISSUE #2756: [7] Account Selector should not auto-match on type-ahead version
it's not a good design to auto list some candidates to-account while user type to-account

image

because of this, I have wrongly transfered to account "alpha" while acctually want to transfer to "alpha-go", several times.

image

alt has also mentioned this in forum

https://bitsharestalk.org/index.php?topic=28250.0

please remove this feature.

and maybe one better solution is to set one white list for each wallet, users can only select accounts in white list to send to.

### PR DESCRIPTION
**Pull request by startailcoon**
This PR closes #2756 (overrides previous PR)

This PR contains a significant larger change to the structure, most notable is the usage of `fetchChainObjects`, recurring quering even if account does not exists and a few code cleanups.

Workings are as follows
- Component mounts and adds favorites and contacts to the searchResult list through `_addToSearch`.
- User types an input and it gets added to teh searchResults list, which will hold `_fetchAccounts` for 250ms to prevent unnecessary queries while the user still inputs data.
- `_fetchAccounts` will query all accounts still missing and process them. Any remaining which have failed (produced no return due to slow node non existing account) will be queried again, up to `max_fails`.
- While query is running, a spinner will be visible at the end of the input.

### Changelog

```
+ Query List
+ Query Cache w/ max query limitor
+ Require active select on Typeahead
+ Loading indicator when query running
+ Delayed querying on input
* Condensed code
* Removed unused code
* Fetch array of objects instead of single
* Results object creation changes
* Changes to error handling
* AccountImage default sizes

### CODE REVIEW

**Comment by startailcoon** [comment]
This PR does not update account fetching on any other component, only Account Selector.

File: app/components/Account/AccountSelector.jsx [hunk 1]
```diff
@@ -94,15 +105,189 @@ class AccountSelector extends React.Component {
         }
     }
 
+    _addToIndex(accountName, noDelay = false) {
```
Comment (sschiessl-bcp) [inline app/components/Account/AccountSelector.jsx]: Please see my change above. Now test this and put `noDelay` to `true`. 

The AccountSelector breaks due to infinity loop, which should not happen. This indicates the rendering / lazy loading is not coupled properly. Please investigate

Comment (startailcoon) [reply app/components/Account/AccountSelector.jsx]: Will look deeper into this, sorry for the miss.

**Comment by startailcoon** [comment]
@sschiessl-bcp removed some unnecessary state changes and added a re-fetch delay. I've tested the update frequency after this, and it seems to be much better.

**Comment by sschiessl-bcp** [comment]
refactored the shit out of it and got lazy loading to work and fixed the amount selector styling
image

**Pull request merged**